Source/WebKit/WebProcess/Extensions/Cocoa/_WKWebExtensionWebRequestFilter.h

```diff
@@ -0,0 +1,41 @@
+// Copyright © 2020 Apple Inc. All rights reserved.
+
+#import <WebKit/WKFoundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class _WKResourceLoadInfo;
+
+// https://developer.chrome.com/extensions/webRequest#type-ResourceType
+typedef NS_ENUM(NSInteger, _WKWebExtensionWebRequestResourceType) {
+    _WKWebExtensionWebRequestResourceTypeMainFrame,
+    _WKWebExtensionWebRequestResourceTypeSubframe,
+    _WKWebExtensionWebRequestResourceTypeStylesheet,
+    _WKWebExtensionWebRequestResourceTypeScript,
+    _WKWebExtensionWebRequestResourceTypeImage,
+    _WKWebExtensionWebRequestResourceTypeFont,
+    _WKWebExtensionWebRequestResourceTypeObject,
+    _WKWebExtensionWebRequestResourceTypeXMLHTTPRequest,
+    _WKWebExtensionWebRequestResourceTypePing,
+    _WKWebExtensionWebRequestResourceTypeCSPReport,
+    _WKWebExtensionWebRequestResourceTypeMedia,
+    _WKWebExtensionWebRequestResourceTypeWebsocket,
+    _WKWebExtensionWebRequestResourceTypeApplicationManifest,
+    _WKWebExtensionWebRequestResourceTypeXSLT,
+    _WKWebExtensionWebRequestResourceTypeBeacon,
+    _WKWebExtensionWebRequestResourceTypeOther,
+};
+
+WK_EXTERN _WKWebExtensionWebRequestResourceType _WKWebExtensionWebRequestResourceTypeFromWKResourceLoadInfo(_WKResourceLoadInfo *);
+
+// https://developer.chrome.com/extensions/webRequest#type-RequestFilter
```

Comment:
Link to the MDN docs when possible.

https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/RequestFilter